### PR TITLE
add kubernetes engine

### PR DESCRIPTION
**Pull request by mceloud**

### CODE REVIEW

File: etcd/server.sls [hunk 1]
```diff
@@ -5,6 +5,31 @@ etcd_packages:
   pkg.installed:
   - names: {{ server.pkgs }}
 
+{%- if server.engine == 'kubernetes' %}
```
Comment (fpytloun) [inline etcd/server.sls]: It would be good to have some default (systemd) in this conditional

Comment (mceloud) [reply etcd/server.sls]: No need. Default value is in metadata.

Comment (fpytloun) [reply etcd/server.sls]: Metadata is reclass-specific. That's why defaults should be also covered in states or map.jinja - to work with non-reclass users (and possibly us if we decide to drop it once).

Comment (mceloud) [reply etcd/server.sls]: Was this decided by community?

Comment (pupapaik) [reply etcd/server.sls]: Filip, reclass is still primary "community" model.

Comment (mceloud) [reply etcd/server.sls]: I did it in @fpytloun way in this case.

**Pull request merged**